File: Apps/BankBase/backend/src/main/java/org/d3javu/backend/model/card/CardType.java
package org.d3javu.backend.model.card;

public enum CardType {
    credit("Кредитная карта"),
    debit("Дебетовая карта"),
    overdraft("Карта с овердрафтом"),
    prepaid("Предоплаченая карта");

    private final String CardType;

    CardType(String defaultName){
        this.CardType = defaultName;
    }

    public String getDefaultCardName() {
        return CardType;
    }
}

File: Apps/BankBase/backend/src/main/scala/model/card/CardType.scala
package model.card

import zio.json.{DeriveJsonCodec, JsonCodec}

enum CardType(val defaultCardName: String) {
  case CREDIT extends CardType("Кредитная карта")
  case DEBIT extends CardType("Дебетовая карта")
  case OVERDRAFT extends CardType("Карта с овердрафтом")
  case PREPAID extends CardType("Предоплаченная карта")
}

object CardType {
  inline given JsonCodec[CardType] = DeriveJsonCodec.gen[CardType]
}